### ISSUE TITLE
Why freeze the batch normalization layer

### ISSUE THREAD

**Issue by yijingru** [yhenon/pytorch-retinanet#24]
Hi, why you freeze the batch normalization layer in codes? Is it about the batch size?

**Comment by yhenon** [comment]
Hi,
> Is it about the batch size?


It is indeed. Because of the memory requirements, the batch size is typically quite low on a standard GPU. This means that batch normalization won't work very well (as it requires a larger batch size). Since the network is pre-trained on imagenet, freezing the weights is a good fix (this is the approach taken in faster-rcnn and others).

**Comment by feixiangdekaka** [comment]
> Hi,
> 
> > Is it about the batch size?
> 
> It is indeed. Because of the memory requirements, the batch size is typically quite low on a standard GPU. This means that batch normalization won't work very well (as it requires a larger batch size). Since the network is pre-trained on imagenet, freezing the weights is a good fix (this is the approach taken in faster-rcnn and others).

when using SGD  ,how can i freeze the bn?
